File: mesmaeker-java-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/MockitoOngoingVoidClause.java
package fr.xgouchet.mesmaeker.mockito;

import org.mockito.Mockito;

import java.util.function.Consumer;

/**
 * @author Xavier F. Gouchet
 */
public class MockitoOngoingVoidClause<T> implements OngoingVoidClause<T> {


    private final T subject;

    public MockitoOngoingVoidClause(T subject) {
        this.subject = subject;
    }

    @Override public OngoingVoidClause<T> when(Consumer<T> consumer) {
        T stubber = Mockito.doNothing().when(subject);
        consumer.accept(stubber);
        return this;
    }
}

File: mesmaeker-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/MockitoOngoingVoidClause.kt
package fr.xgouchet.mesmaeker.mockito

import org.mockito.Mockito

/**
 * @author Xavier F. Gouchet
 */
class MockitoOngoingVoidClause<T>(private val subject: T) : OngoingVoidClause<T> {

    override fun `when`(block: (T) -> Unit): OngoingVoidClause<T> {
        return whenever(block)
    }

    override fun whenever(block: (T) -> Unit): OngoingVoidClause<T> {
        val stubber = Mockito.doNothing().`when`(subject)
        block(stubber)
        return this
    }
}
